src/main/java/org/opensearch/neuralsearch/query/HybridQueryScorer.java

```diff
@@ -111,24 +112,34 @@ public float[] hybridScores() throws IOException {
         DisiWrapper topList = subScorersPQ.topList();
         for (DisiWrapper disiWrapper = topList; disiWrapper != null; disiWrapper = disiWrapper.next) {
             // check if this doc has match in the subQuery. If not, add score as 0.0 and continue
-            if (disiWrapper.scorer.docID() == DocIdSetIterator.NO_MORE_DOCS) {
+            Scorer scorer = disiWrapper.scorer;
+            if (scorer.docID() == DocIdSetIterator.NO_MORE_DOCS) {
                 continue;
             }
-            float subScore = disiWrapper.scorer.score();
-            scores[queryToIndex.get(disiWrapper.scorer.getWeight().getQuery())] = subScore;
+            Query query = scorer.getWeight().getQuery();
+            List<Integer> indexes = queryToIndex.get(query);
+            // we need to find the index of first sub-query that hasn't been updated yet
+            int index = indexes.stream()
+                .mapToInt(idx -> idx)
+                .filter(index1 -> Float.compare(scores[index1], 0.0f) == 0)
+                .findFirst()
+                .orElseThrow(() -> new IllegalStateException("cannot collect score for subquery"));
```

Comment:
Please make the exception error message more verbose.

Reply:
ack